app/scripts/controllers/inventory/viewStockLevels.js

```diff
@@ -0,0 +1,338 @@
+            bold: true
+          }
+        );
+      } else {
+        var selectedDistributionCenter = $scope.distributionCenters.filter(function(center) {
+          return $scope.searchedParams.distributionCenter === center.id.toString();
+        });
+        prefix.push(
+          {
+            text: 'Distribution Center: ',
+            fontSize: 8,
+            bold: true
+          }, {
+            text: selectedDistributionCenter[0].name
+          }
+        );
+      }
+
+      docDefinition.content = [{text: prefix, margin: [-10, 0, 0, 0], fontSize: 7}].concat(docDefinition.content);
+      return docDefinition;
+    }
+
+    function exportPDFHeader() {
+      return [
+        {
+          text: 'Stock Levels: ' + (isInStockSearch() ? 'In Stock' : 'Not Labelled'),
+          fontSize: 10,
+          bold: true,
+          margin: [30, 20, 0, 0]
+        }
+      ];
+    }
+
+    function exportPDFFooter(gridOptions, currentPage, pageCount) {
+      var columns = [
+        {text: 'Date generated: ' + $filter('bsisDateTime')(new Date()), width: 'auto'},
+        {text: 'Page ' + currentPage + ' of ' + pageCount, style: {alignment: 'right'}}
+      ];
+      return {
+        columns: columns,
+        columnGap: 10,
+        margin: [30, 0]
+      };
+    }
+
+    $scope.gridOptions = {
+      data: [],
+      columnDefs: [],
+
+      exporterPdfOrientation: 'portrait',
+      exporterPdfPageSize: 'A4',
+      exporterPdfDefaultStyle: {fontSize: 6, margin: [-2, 0, 0, 0] },
+      exporterPdfTableHeaderStyle: {fontSize: 6, bold: true, margin: [-2, 0, 0, 0] },
+      exporterPdfMaxGridWidth: 400,
+
+      exporterPdfCustomFormatter: function(docDefinition) {
+        return exportPDFCustomFormatter(docDefinition);
+      },
+      exporterPdfHeader: function() {
+        return exportPDFHeader();
+      },
+      exporterPdfFooter: function(currentPage, pageCount) {
+        return exportPDFFooter($scope.gridOptions, currentPage, pageCount);
+      },
+      onRegisterApi: function(gridApi) {
+        $scope.gridApi = gridApi;
+      }
+    };
+
+    $scope.export = function(format) {
+      if (format === 'pdf') {
+        $scope.gridApi.exporter.pdfExport('all', 'all');
+      } else if (format === 'csv') {
+        $scope.gridApi.exporter.csvExport('all', 'all');
+      }
+    };
+
+    $scope.clearSearch = function(searchForm) {
+      $scope.search = angular.copy(master);
+      $scope.searched = false;
+      searchForm.$setPristine();
+      $location.search({});
+    };
+
+    $scope.clearDistributionCenter = function() {
+      $scope.search.distributionCenter = null;
+    };
+
+    $scope.updateAllSites = function() {
+      if ($scope.search.distributionCenter) {
+        $scope.search.allSites = false;
+      } else {
+        $scope.search.allSites = true;
+      }
+    };
+
+    $scope.viewStockLevels = function(searchForm) {
+      if (searchForm.$invalid) {
+        return;
+      }
+      $scope.searched = false;
```

Comment:
I tried to replace `$scope.searched` with `searchForm.$submitted` however because the searched scope variable is used to toggle the results table what happens is that the table is shown (empty) if the form has been submitted but is invalid. I then tried using `searchForm.$submitted && !searchForm.$invalid` in the `ng-if` condition for the table and an empty table is then displayed as soon as the error in the form has been cleared (not after the submit button is pressed again).